nntrainer/layers/layer_context.h

```diff
@@ -525,6 +525,22 @@ class RunLayerContext {
    */
   const std::string &getName() const { return std::get<props::Name>(props); }
 
+  /**
+   * @brief   check if run context is set and is ready to use
+   *
+   * @return true if ready, else false
+   */
+  bool readyToUse() const {
+    /**
+     * assumption:
+     * 1. there must be atleast 1 input
+     * 2. the setter set everything at once
+     */
+    if (inputs.empty())
+      return false;
+    return !inputs[0]->getVariable().uninitialized();
```

Comment:
Is checking for the first input enough?

Reply:
If assumption holds, it looks safe

Reply:
yes, if the assumption stays valid. Even if we check all, and the assumption does not hold true, some required inputs might not be set, and checking all inputs wont be enough either.